arch/src/aarch64/fdt.rs

```diff
@@ -233,29 +233,33 @@ fn create_memory_node(
 ) -> FdtWriterResult<()> {
     // See https://github.com/torvalds/linux/blob/master/Documentation/devicetree/booting-without-of.txt#L960
     // for an explanation of memory nodes.
-
     if numa_nodes.len() > 1 {
         for numa_node_idx in 0..numa_nodes.len() {
             let numa_node = numa_nodes.get(&(numa_node_idx as u32));
+            let mut mem_reg_prop: Vec<u64> = Vec::new();
+            let mut node_memory_addr: u64 = 0;
             // Each memory zone of numa will have its own memory node, but
             // different numa nodes should not share same memory zones.
             for memory_region in numa_node.unwrap().memory_regions.iter() {
                 let memory_region_start_addr: u64 = memory_region.start_addr().raw_value();
                 let memory_region_size: u64 = memory_region.size() as u64;
-                let mem_reg_prop = [memory_region_start_addr, memory_region_size];
-                // With feature `acpi` enabled, RAM at 0-4M is for edk2 only
-                // and should be hidden to the guest.
+                // RAM at 0-4M is hidden to the guest for edk2
```

Comment:
I think you can just simply drop the `aarch64: Update NUMA node in FDT for memory hole` commit? or just change the comments here because we don't have feature `acpi` anymore. I understand you want to group all memory in a NUMA node to a single memory node, but if there is a memory hole in this NUMA node, for example:
```
memory@bfc00000/
    device_type: "memory"
    reg: [0, bfc00000, 0, 3c400000, 1, 0, 0, 3c00000]
    numa-node-id: [2]
```
The `reg: [0, bfc00000, 0, 3c400000, 1, 0, 0, 3c00000]` actually indicates two memory regions, one starts from 0xbfc00000 and one starts from 0x100000000, in this case the meaning of this device tree memory node `memory@bfc00000` does not make sense anymore, so I would say it is better to use two memory nodes `memory@bfc00000` and `memory@100000000` instead of the current one.

In this case, the refactoring you did in this commit is less meaningful to me..

Reply:
> I think you can just simply drop the `aarch64: Update NUMA node in FDT for memory hole` commit? or just change the comments here because we don't have feature `acpi` anymore. I understand you want to group all memory in a NUMA node to a single memory node, but if there is a memory hole in this NUMA node, for example:
> 
> ```
> memory@bfc00000/
>     device_type: "memory"
>     reg: [0, bfc00000, 0, 3c400000, 1, 0, 0, 3c00000]
>     numa-node-id: [2]
> ```
> 
> The `reg: [0, bfc00000, 0, 3c400000, 1, 0, 0, 3c00000]` actually indicates two memory regions, one starts from 0xbfc00000 and one starts from 0x100000000, in this case the meaning of this device tree memory node `memory@bfc00000` does not make sense anymore, so I would say it is better to use two memory nodes `memory@bfc00000` and `memory@100000000` instead of the current one.
> 
> In this case, the refactoring you did in this commit is less meaningful to me..

The 2 ways of creating the `memory` node are equally correct. The device tree spec (https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3) describes about it (Chapter 3.4), and give examples for the exactly same case. 

Your example means: a memory node starts from bfc00000, 2 regions are in it, and there is a hole. Isn't it the exact thing we want to describe?

The additional advantage of the refactoring is: The device tree is more compact, a numa node is described by one `memory` node only.

Reply:
ok